projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -342,6 +349,8 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         if (value && value.length > 10) {
             throw Error('Maximum amount of grouped columns is 10.');
         }
+        const oldExpressions: Array<ISortingExpression> = this.groupingExpressions;
+        const newExpressions: Array<ISortingExpression> = value;
         this._groupingExpressions = cloneArray(value);
```

Comment:
This is not part of the change, but why is a deep copy of the array provided needed?

Reply:
@kdinev If you refer to this line : `this._groupingExpressions = cloneArray(value);` - it doesn't create a deep copy, because the optional boolean parameter hasn't been set to "true".

Reply:
Ok, then why is a new reference created is the question? Why can't `value` be used directly?